lib/light-service/context/key_verifier.rb

```diff
@@ -40,13 +40,13 @@ def verify
       context
     end
 
-    def self.verify_keys(context, &block)
-      ReservedKeysVerifier.new(context).verify
-      ExpectedKeyVerifier.new(context).verify
+    def self.verify_keys(context, action, &block)
+      ReservedKeysVerifier.new(context, action).verify
+      ExpectedKeyVerifier.new(context, action).verify
 
       block.call
 
-      PromisedKeyVerifier.new(context).verify
+      PromisedKeyVerifier.new(context, action).verify
     end
```

Comment:
@adomokos if I understand you right:

When you change the verifiers to accept `context.current_action` as the second argument, then all the actions share a state. This has the same effect as just passing `context` and extracting the `current_action` from the context. Whenever there's a change to `context.current_action`, the verifiers become affected.

This pull request keeps that state away from the key verification process. Yes, `context` still needs to be passed because that's where the keys live, but the state remains unchanged during the key verification of that action.

I hope I answered your question.